plugins/woocommerce/client/admin/client/settings-payments/settings-payments-main.tsx

```diff
@@ -36,45 +33,57 @@ export const SettingsPaymentsMain = () => {
 		return select( PLUGINS_STORE_NAME ).getInstalledPlugins();
 	}, [] );
 
-	useEffect( () => {
-		setWooPaymentsGatewayData(
-			parseScriptTag( 'experimental_wc_settings_payments_woopayments' )
-		);
-		setRegisteredPaymentGateways(
-			parseScriptTag( 'experimental_wc_settings_payments_gateways' )
-		);
-		setPreferredPluginSuggestions(
-			parseScriptTag(
-				'experimental_wc_settings_payments_preferred_extensions_suggestions'
-			)
-		);
-		setOtherPluginSuggestions(
-			parseScriptTag(
-				'experimental_wc_settings_payments_other_extensions_suggestions'
-			)
-		);
-	}, [] );
+	// Make UI to refresh when plugin is installed.
+	const { invalidateResolutionForStoreSelector } = useDispatch(
+		PAYMENT_SETTINGS_STORE_NAME
+	);
+
+	const {
+		registeredPaymentGateways,
+		preferredPluginSuggestions,
+		otherPluginSuggestions,
+	} = useSelect( ( select ) => {
+		return {
+			registeredPaymentGateways: select(
+				PAYMENT_SETTINGS_STORE_NAME
+			).getRegisteredPaymentGateways(),
+			preferredPluginSuggestions: select(
+				PAYMENT_SETTINGS_STORE_NAME
+			).getPreferredExtensionSuggestions(),
+			otherPluginSuggestions: select(
+				PAYMENT_SETTINGS_STORE_NAME
+			).getOtherExtensionSuggestions(),
+		};
+	} );
 
 	const setupPlugin = useCallback(
-		( plugin: Plugin ) => {
+		( extension: SuggestedPaymentExtension ) => {
 			if ( installingPlugin ) {
 				return;
 			}
-			setInstallingPlugin( plugin.id );
-			installAndActivatePlugins( [ plugin.plugins[ 0 ] ] )
+			setInstallingPlugin( extension.id );
+			installAndActivatePlugins( [ extension.plugin.slug ] )
 				.then( ( response ) => {
 					createNoticesFromResponse( response );
-					// Reload the page to reflect the changes - will be replaced with a more elegant solution in the future.
-					window.location.reload();
+					invalidateResolutionForStoreSelector(
+						'getRegisteredPaymentGateways'
+					);
+					setInstallingPlugin( null );
 				} )
 				.catch( ( response: { errors: Record< string, string > } ) => {
 					createNoticesFromResponse( response );
 					setInstallingPlugin( null );
 				} );
 		},
-		[ installAndActivatePlugins, installingPlugin ]
+		[ installingPlugin, installAndActivatePlugins ]
```

Comment:
Should we also add `invalidateResolutionForStoreSelector` to dependency array?

Reply:
yes! great catch, I've missed it